server.go

```diff
@@ -677,7 +677,14 @@ func (s *Server) handleAutofill(links []string, w http.ResponseWriter, r *http.R
 					// Get Data from MAL (jikan api)
 					rgx := regexp.MustCompile(`[htp]{4}s?:\/\/[^\/]*\/anime\/([0-9]*)`)
 					match := rgx.FindStringSubmatch(sourcelink)
-					id := match[1]
+					var id string
+					if len(match) < 1 {
```

Comment:
This should be < 2.  The first element is the whole match of the regex, second element is the first submatch.  The second element is used to get the ID below.

Reply:
Fixed with b640bb5